### PR TITLE
Enable nGraph GPU transformer

### PR DESCRIPTION
**Pull request by csullivan**
## Description ##
Enables the nGraph GPU transformer in MXNet.

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- Added FComputeEx<gpu> for forward and backward compute
- Removed multicontext conditionals that prevent GPU transformer from being utilized in graph_executor.cc

## Comments ##
- Build configuration used (but not committed): https://gist.github.com/csullivan/363b30c36804060cb33f3077540b6ec3

### CODE REVIEW

File: src/ngraph/ngraph_nnvm_ops.cc [hunk 1]
```diff
@@ -259,15 +259,17 @@ void register_forward_op(std::shared_ptr<Graph> graph) {
                                               dispatch_mode,
                                               mxnet::DispatchMode::kFComputeEx);
       });
-  // create the compute lambda
-  op.set_attr<mxnet::FComputeEx>(
-      "FComputeEx<cpu>",
+  // create the cpu & gpu forward compute lambdas
+  for (std::string arch : {"cpu", "gpu"}) {
+    op.set_attr<mxnet::FComputeEx>(
+      "FComputeEx<" + arch + ">",
```
Comment (mbrookhart) [inline src/ngraph/ngraph_nnvm_ops.cc]: This is nice. 👍

File: src/executor/graph_executor.cc [hunk 2]
```diff
@@ -562,16 +562,13 @@ void GraphExecutor::Init(nnvm::Symbol symbol,
                             arg_grad_ctxes, aux_state_ctxes, grad_req_types);
 
 #if MXNET_USE_NGRAPH == 1
-  // TODO(mbrookhart): Remove this when hetr can handle multiple contexts
-  auto multi_context = multi_context_check(default_ctx, in_arg_ctxes,
-                                           arg_grad_ctxes, aux_state_ctxes);
```
Comment (mbrookhart) [inline src/executor/graph_executor.cc]: We need to keep this is place, we can't handle multiple contexts, this will break other unit tests.

Comment (csullivan) [reply src/executor/graph_executor.cc]: As suggested, this change has been replaced by adding an env. variable MXNET_NGRAPH_GPU which toggles the conditional at L525 when set.

**Comment by csullivan** [comment]
TESTNOW

**Comment by mbrookhart** [comment]
TESTNOW

**Comment by csullivan** [comment]
TESTNOW

**Comment by csullivan** [comment]
TESTNOW

**Comment by csullivan** [comment]
TESTNOW

**Comment by csullivan** [comment]
TESTNOW

**Comment by csullivan** [comment]
TESTNOW

**Pull request merged**